packages/firebase_analytics/test/firebase_analytics_test.dart

```diff
@@ -117,36 +108,30 @@ void main() {
   });
 
   group('$FirebaseAnalytics analytics events', () {
-    FirebaseAnalytics analytics;
-
     String name;
     Map<String, dynamic> parameters;
 
     setUp(() {
-      final MockPlatformChannel mockChannel = MockPlatformChannel();
-
       name = null;
       parameters = null;
 
-      // TODO(amirh): remove this on when the invokeMethod update makes it to stable Flutter.
-      // https://github.com/flutter/flutter/issues/26431
-      // ignore: strong_mode_implicit_dynamic_method
-      when(mockChannel.invokeMethod('logEvent', any))
-          .thenAnswer((Invocation invocation) {
-        final Map<String, dynamic> args = invocation.positionalArguments[1];
+      channel.setMockMethodCallHandler((MethodCall call) async {
+        if (call.method != 'logEvent') {
+          throw ArgumentError('Only logEvent invocations expected');
+        }
+        final Map<String, dynamic> args =
+            Map<String, dynamic>.from(call.arguments);
         name = args['name'];
-        parameters = args['parameters'];
+        if (args['parameters'] != null) {
```

Comment:
Should we first set `parameters = null` to make sure we're not left with a stale value from a previous invocation?

Reply:
ok, agree better safe than sorry. I'll make the change in a moment.

Reply:
resolve with latest push